src/makeInfinityAdder.js

```diff
@@ -4,7 +4,22 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(num) {
+    if (num === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
+    sum += num;
```

Comment:
Ensure that `num` is a number before adding it to `sum`. Consider adding a check to handle non-numeric inputs gracefully, as adding non-numeric values could lead to unexpected results.